src/variables/body/search.tsx

```diff
@@ -92,45 +95,62 @@ const useOutsideClick = (ref: any, callback: any) => {
 };
 
 const ScopeMenuComponent = ({ model }: { model: Variables.IModel }) => {
-  const [toggleState, setToggle] = useState(false);
+  // const [toggleState, setToggle] = useState(false);
+  // const [scopes, setScopes] = useState(model.scopes);
   const [scope, setScope] = useState(model.currentScope);
   const wrapperRef = useRef(null);
-  const scopes = model.scopes;
+
+  useEffect(() => {
+    const updateScopes = (_: Variables.IModel, updates: Variables.IScope[]) => {
+      const scope = !!updates && updates.length > 0 ? updates[0] : null;
+      // setScopes(updates);
+      setScope(scope);
+    };
+    model.scopesChanged.connect(updateScopes);
+
+    return () => {
+      model.scopesChanged.disconnect(updateScopes);
+    };
+  });
 
   const onClickOutSide = () => {
-    setToggle(false);
+    // setToggle(false);
   };
 
-  const toggle = (e: any) => {
-    setToggle(!toggleState);
+  const toggle = () => {
+    // if (!!scopes) {
+    //   setToggle(!toggleState);
+    // }
   };
 
   useOutsideClick(wrapperRef, onClickOutSide);
 
-  const changeScope = (newScope: Variables.IScope) => {
-    if (newScope === scope) {
-      return;
-    }
-    setScope(newScope);
-    model.currentScope = newScope;
-    setToggle(false);
-  };
-
-  const List = (
-    <ul>
-      {scopes.map(scope => (
-        <li key={scope.name} onClick={e => changeScope(scope)}>
-          {scope.name}
-        </li>
-      ))}
-    </ul>
-  );
+  // const changeScope = (newScope: Variables.IScope) => {
+  //   if (newScope === scope) {
+  //     return;
+  //   }
+  //   setScope(newScope);
+  //   model.currentScope = newScope;
+  //   setToggle(false);
+  // };
+
+  // const List = (
+  //   <ul>
+  //     {!!scopes
+  //       ? scopes.map(scope => (
+  //           <li key={scope.name} onClick={e => changeScope(scope)}>
+  //             {scope.name}
+  //           </li>
+  //         ))
+  //       : null}
+  //   </ul>
+  // );
```

Comment:
Any reason for keeping these comments around?

Reply:
I forgot about this. Sry

Reply:
:+1: